pkgs/top-level/all-packages.nix

```diff
@@ -3085,6 +3085,8 @@ with pkgs;
 
   artyFX = callPackage ../applications/audio/artyFX { };
 
+  astro-language-server = callPackage ../by-name/as/astro-language-server/package.nix { pnpm_8 = pnpm_9; };
```

Comment:
Why not just change the dependency in the package.nix?

Reply:
https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults